inference/core/interfaces/stream/model_handlers/workflows.py

```diff
@@ -25,15 +25,11 @@ def run_workflow(
             except:
                 event_loop = asyncio.new_event_loop()
             self._event_loop = event_loop
-        if len(video_frames) > 1:
-            logger.warning(
-                f"Workflows in InferencePipeline do not support multiple video sources. Using the first video frame."
-            )
         if workflows_parameters is None:
             workflows_parameters = {}
-        workflows_parameters[image_input_name] = video_frames[0].image
-        return [
-            execution_engine.run(
-                runtime_parameters=workflows_parameters, event_loop=self._event_loop
-            )
+        workflows_parameters[image_input_name] = [
+            video_frame.image for video_frame in video_frames
```

Comment:
woah, this means we get parallel video processing on same workflow via batch?

Reply:
not everything will work (like Roboflow sink due to asyncio), but yes